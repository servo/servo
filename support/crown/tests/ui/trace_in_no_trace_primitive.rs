/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
//@rustc-env:RUSTC_BOOTSTRAP=1

/// Mock `JSTraceable`
pub trait JSTraceable {}
impl JSTraceable for i32 {}

// second generic argument must not be traceable
#[crown::trace_in_no_trace_lint::must_not_have_traceable(0)]
struct NoTrace<NoTraceable> {
    n: NoTraceable,
}

struct Foo(NoTrace<i32>);

fn main() {}
