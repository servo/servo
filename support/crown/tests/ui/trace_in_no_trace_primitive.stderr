warning: must_not_have_traceable marked wrapper is not needed for types that implements empty Traceable (like primitive types). Consider removing the wrapper.
  --> $DIR/trace_in_no_trace_primitive.rs:16:12
   |
16 | struct Foo(NoTrace<i32>);
   |            ^^^^^^^^^^^^
   |
   = note: `#[warn(crown::empty_trace_in_no_trace)]` on by default

warning: 1 warning emitted

